apps/wallet-mobile/src/TxHistory/ActionsBanner.tsx

```diff
@@ -58,6 +58,23 @@ export const ActionsBanner = ({disabled = false}: {disabled: boolean}) => {
             <Text style={styles.actionLabel}>{strings.receiveLabel}</Text>
           </View>
 
+          {features.showSwapButton && <Spacer width={32} />}
+
+          {features.showSwapButton && (
+            <View style={styles.centralized}>
+              <TouchableOpacity
+                style={styles.actionIcon}
+                onPress={navigateTo.swap}
+                testID="receiveButton"
+                disabled={disabled}
+              >
+                <Icon.Received {...ACTION_PROPS} />
+              </TouchableOpacity>
+
+              <Text style={styles.actionLabel}>SWAP</Text>
```

Comment:
requires translations
```suggestion
              <Text style={styles.actionLabel}>{strings.swap}</Text>
```